frontend/app/features/password-reset/components/Reset-Password.tsx

```diff
@@ -0,0 +1,73 @@
+"use client";
+
+import Button from "@components/button/Button";
+import Input from "@components/input/Input";
+import { useRouter, useSearchParams } from "next/navigation"; 
+import { useState } from "react";
+import styles from "../styles/resetPassword.module.css";
+
+const ResetPasswordPage = () => {
+  const searchParams = useSearchParams();
+  const token = searchParams ? searchParams.get("token") : null;
```

Comment:
if token is not present, can you redirect the user to a not found page?, actually it would be better to validate the token first as well, but in this scenario, its not needed, just check if its there, if not , redirect to not found page